src/content/capture/video-contenthint/js/main.js

```diff
@@ -0,0 +1,120 @@
+/*
+*  Copyright (c) 2016 The WebRTC project authors. All Rights Reserved.
+*
+*  Use of this source code is governed by a BSD-style license
+*  that can be found in the LICENSE file in the root of the source
+*  tree.
+*/
+
+'use strict';
+
+var srcVideo = document.getElementById('srcVideo');
+var fluidVideo = document.getElementById('fluidVideo');
+var detailedVideo = document.getElementById('detailedVideo');
+
+var srcStream;
+var fluidStream;
+var detailedStream;
+
+var offerOptions = {
+  offerToReceiveAudio: 0,
+  offerToReceiveVideo: 1
+};
+
+function maybeCreateStream() {
+  if (srcStream) {
+    return;
+  }
+  if (srcVideo.captureStream) {
+    srcStream = srcVideo.captureStream();
+    call();
+  } else {
+    trace('captureStream() not supported');
+  }
+}
+
+// Video tag capture must be set up after video tracks are enumerated.
+srcVideo.oncanplay = maybeCreateStream;
+if (srcVideo.readyState >= 3) {  // HAVE_FUTURE_DATA
+  // Video is already ready to play, call maybeCreateStream in case oncanplay
+  // fired before we registered the event handler.
+  maybeCreateStream();
+}
+
+srcVideo.play();
+
+function setVideoTrackContentHints(stream, hint) {
+  var tracks = stream.getVideoTracks();
+  tracks.forEach(function(track) {
+    if ('contentHint' in track) {
+      track.contentHint = hint;
+      if (track.contentHint !== hint) {
+        trace('Invalid video track contentHint: \'' + hint + '\'');
+      }
+    } else {
+      trace('MediaStreamTrack contentHint attribute not supported');
+    }
+  });
+}
+
+function call() {
+  // This creates multiple independent PeerConnections instead of multiple
+  // streams on a single PeerConnection object so that b=AS (the bitrate
+  // constraints) can be applied independently.
+  fluidStream = srcStream.clone();
+  setVideoTrackContentHints(fluidStream, 'fluid');
+  establishPC(fluidVideo, fluidStream);
+  detailedStream = srcStream.clone();
+  setVideoTrackContentHints(detailedStream, 'detailed');
+  establishPC(detailedVideo, detailedStream);
+}
+
+function establishPC(videoTag, stream) {
+  var pc1 = new RTCPeerConnection(null);
+  var pc2 = new RTCPeerConnection(null);
+  pc1.onicecandidate = function(e) {
+    onIceCandidate(pc1, pc2, e);
+  };
+  pc2.onicecandidate = function(e) {
+    onIceCandidate(pc2, pc1, e);
+  };
+  pc2.onaddstream = function(event) {
+    videoTag.srcObject = event.stream;
+  };
+
+  pc1.addStream(stream);
+  pc1.createOffer(offerOptions).then(
+    function(desc) {
+      pc1.setLocalDescription(desc).catch(onSetSessionDescriptionError);
+      pc2.setRemoteDescription(desc).catch(onSetSessionDescriptionError);
+      // Since the 'remote' side has no media stream we need
+      // to pass in the right constraints in order for it to
+      // accept the incoming offer of audio and video.
+      pc2.createAnswer().then(
```

Comment:
You ought to chain createAnswer() off setRemoteDescription, like this:

pc2.setRemoteDescription(desc).then(function(){pc2.createAnswer()}).then(function(answerDesc)....

(suitably line-broken and indented).
A then-function should usually return a promise, which can then be passed to the next .then-function.